File: pset2/vigenere/vigenere.c
// Encrypt a word using Vigenere's cipher.
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <cs50.h>

int main(int argc, string argv[])
{
    // Ensure the arg alphabetic word is inputted, otherwise return error.
    if (argc != 2)
    {
        printf("Error, please input only alphabetic word.\n");
        return 1;
    }

    // Ensure the arg alphabetic word, else return an error and exit.
    for (int i = 0; i < strlen(argv[1]); i++)
    {
        if (!isalpha(argv[1][i]))
        {
            printf("Error, please input only alphabetic word.\n");
            return 1;
        }
    }
    string key = argv[1];
    string plaintext = get_string("plaintext: ");
    printf("ciphertext: ");
    for (int i = 0, j = 0; i < strlen(plaintext); i++)
    {
        int letterKey = tolower(key[j % strlen(key)]) - 'a';
        if (isalpha(plaintext[i]))
        {
            char plainchar = plaintext[i];
            if (isupper(plainchar))
                plainchar = (((plainchar + letterKey) - 65 ) % 26) + 65;
            if (islower(plainchar))
                plainchar = (((plainchar + letterKey) - 97 ) % 26) + 97;
            plaintext[i] = plainchar;
            j++;
        }
    printf("%c", plaintext[i]);
    }
    printf("\n");
    return 0;
}

File: pset6/vigenere/vigenere.py
# Port of caesar.c to Python
import cs50
from sys import argv

def main():
    # Ensure the user entered at least one arg, otherwise exit
    if (len(argv) != 2):
        print("Usage: python caesar.py <integer>")
        exit(1)

    if argv[1].isalpha() == False:
        print("Error, input only alphabeticword.")
        exit(1)

    key = argv[1]
    cipher = []
    print("plaintext: ", end="")
    plaintext = cs50.get_string()
    print("ciphertext: ", end="")
    j = 0

    for c in plaintext:
        if c.isalpha():
            letterKey = ord(key[j % len(key)].lower()) - 97
            j = j + 1
            cipher.append(caesar(c, letterKey))
        else:
            cipher.append(c)

    for c in cipher:
        print(c, end="")
    print()
    exit(0)

def caesar(char, key):
    if char.isupper():
        return chr((((ord(char) + key) - 65 ) % 26) + 65)
    else:
        return chr((((ord(char) + key) - 97 ) % 26) + 97)

if __name__ == "__main__":
    main()